validator/src/main/java/com/google/daq/mqtt/util/ConfigManager.java

```diff
@@ -311,4 +316,42 @@ private DiscoveryConfig getDiscoveryConfig() {
     return discoveryConfig;
   }
 
+  public List<Exception> getSchemaViolations() {
+    return schemaViolations;
+  }
+
+  private void recordIfSchemaViolations(Runnable function) {
+    try {
+      function.run();
+    } catch (Exception e) {
+      schemaViolations.add(e);
+    }
+  }
+
+  private <T> void recordIfSchemaViolations(Consumer<T> consumer, T argument) {
```

Comment:
Are these three variants used? I don't see them anywhere... and, even so, could they be collapsed into two forms (supplier or runnable) and have the argument be encapsulated in the calling lambda?